sno/diff.py

```diff
@@ -161,13 +166,28 @@ def diff_with_writer(
         if feature_filter is not UNFILTERED:
             all_datasets = all_datasets.intersection(feature_filter.keys())
 
+        dataset_geometry_transforms = {}
+        if target_srs is not None:
+
+            dest_srs = make_srs(target_srs)
+            for dataset_path in all_datasets:
+                dataset = base_rs.get(dataset_path) or target_rs.get(dataset_path)
+                # is there a better way to do this?
+                gc = dataset.get_meta_item("gpkg_geometry_columns")
```

Comment:
You could add an "srs_id" field to dataset - seems good to have some useful things like that as part of the interface, as opposed to making callers figure out how to extract it from metadata.

There are some examples of these eg "geom_column_name" (defined for both datasets at once in structure.py) and "primary_key" (defined differently for each dataset in dataset1.py and dataset2.py)